### PR TITLE
chore(tests): remove stencilSystem references

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Unit tests (`npm test`) were run locally and passed
- [x] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

GitHub Issue Number: N/A

## What is the new behavior?

this commit renames a helper function, `mockStencilSystem`, to
`mockCompilerSystem`. `StencilSystem` was a type that previously existed
in the codebase until the legacy compiler was refactored in https://github.com/ionic-team/stencil/commit/63595bc540db9aa85295f5a8dc8738beb2a37611 (https://github.com/ionic-team/stencil/commit/63595bc5#diff-c396c2ecfcbdf4562ae518122990719c12b5ed9bd64908ef2142620f6d1440dbR209).

this change is being made to reduce the number of '*System' types that
exist in the codebase, and in an attempt to provide greater clarity
around the hierarchy around `CompilerSystem` types.

this function does keep the wrapper method that is responsible for
generating the `TestingSystem`. the justification (also explained in the
JSDocs for the function) is that this 'felt' like a logical location
that engineers working on the project (a file containing mocks). the
team is currently considering what the best location for all mocks to be
placed is (STENCIL-377), which may result in the function moving (or
being eliminated for the function it wraps).

this commit does change one line in `transpile.ts`, where assignment to
the optional `sys` parameter is removed. because JavaScript is 'calls
by sharing' (https://en.wikipedia.org/wiki/Evaluation_strategy#Call_by_sharing)
the reassignment was never actually being used. a typescript playground
illustrates this: https://www.typescriptlang.org/play?#code/MYewdgzgLgBAZiEMC8MDeBYAUDG4CmAXDAIwA02uUA7iMZjrjFABYBO+RMARKx-twqMAvkOEBubNlCRYAIwCuASwA2AEwBiiFDAAUAShQA+dJRgcoCtmFONcBYgE5nQps1r0zbvp2LcAViAA1gKuTKJmEthRWNhwCmDAUErgMGogAIIAKixKYADmunAA-MRQAJ4ADvggcPCIhgy4MhAgKvgAdCoghQAGAJI2AOTp2bkFQ2QwcvgIHDDU+Ob4AIYQEEr5NqxLlStsKwC2+FD4bDAAJGgAUgDKAPIAch3QbHn5SnDlRVNgCioqKYAFn0wl6+kkjDqqDqAB9YdNlOotCADJDmuBWu0uj1dANhqMcu9JjAVnBTudFss1hstswWLt9kcTmdLjcHs9Xu9Pt84L9-oCYCCwRDolIsC02p1un0AEKzEDzYArAHvGAjTJEiZTHYwABuKoUS1q9SQSggbLuTxeUDeBR5RUQ-IBwNB4MhhPGhVFEsxUpxfQy5NZytVBXVnuJOoZ+sNxrqCDNFquVs5tu5X0dIGdguF7uwQA

this commit removes the unneeded `sys` parameter from the testing
utility `transpileModule`.

this commit removes the final instances of 'stencilSys' by renaming
usages of the string (which was only used as parameter names when
stencil patches typescript) in order to complete the 'migration' away
from the `StencilSys` type

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

Unit tests were run and passed (which is the utility that _uses_ this functionality (that was changed).

Patching TS is a bit trickier to test, which wasn't done as a part of this effort (but we're renaming a parameter on an internal function, so it's _probably_ fine)

### CODE REVIEW

File: src/compiler/transformers/test/transpile.ts [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
+ * @param compilerCtx a compiler context to use in the transpilation process
+ * @param beforeTransformers TypeScript transformers that should be applied before the code is emitted
+ * @param afterTransformers TypeScript transformers that should be applied after the code is emitted
+ * @returns the result of the transpilation step
+ */
 export function transpileModule(
   input: string,
   config?: d.Config,
   compilerCtx?: d.CompilerCtx,
-  sys?: d.CompilerSystem,
```
Comment (rwaskiewicz) [inline src/compiler/transformers/test/transpile.ts:13]: The removal of this parameter is arguably the 'big' change in this PR (and the subsequent assignment being removed). See the PR summary for why i think this is an OK change to make

**Comment by rwaskiewicz** [comment]
The BrowserStack tests appear to be socket/connection related. Marking this as ready to review.

File: src/compiler/transformers/test/transpile.ts (hunk 1 above)
Comment (alicewriteswrongs) [inline src/compiler/transformers/test/transpile.ts:13]: any idea why eslint didn't pick this up as an unused variable?

**Review by alicewriteswrongs — APPROVED** [review]
lgtm!

File: src/compiler/transformers/test/transpile.ts (hunk 1 above)
Comment (rwaskiewicz) [reply src/compiler/transformers/test/transpile.ts:13]: That's a good question 🤔 

I can trigger other lint errors (like our `@param` JSDoc ones) in this file, which tells me the linter is picking up this rule. 

I created a [mini-sample in the ESLint playground](https://eslint.org/demo#eyJ0ZXh0IjoiZnVuY3Rpb24gYmFyKGJheikge1xufVxuXG5mdW5jdGlvbiBiYXIyKGJheikge1xuICBiYXogPSBiYXogfHwge307XG59XG52YXIgZm9vID0gNTtcbmJhcigpO1xuYmFyMihmb28pOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOjIsImZvci1kaXJlY3Rpb24iOjIsImdldHRlci1yZXR1cm4iOjIsIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOjIsIm5vLWNhc2UtZGVjbGFyYXRpb25zIjoyLCJuby1jbGFzcy1hc3NpZ24iOjIsIm5vLWNvbXBhcmUtbmVnLXplcm8iOjIsIm5vLWNvbmQtYXNzaWduIjoyLCJuby1jb25zdC1hc3NpZ24iOjIsIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6Miwibm8tY29udHJvbC1yZWdleCI6Miwibm8tZGVidWdnZXIiOjIsIm5vLWRlbGV0ZS12YXIiOjIsIm5vLWR1cGUtYXJncyI6Miwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjoyLCJuby1kdXBlLWVsc2UtaWYiOjIsIm5vLWR1cGUta2V5cyI6Miwibm8tZHVwbGljYXRlLWNhc2UiOjIsIm5vLWVtcHR5IjoyLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLWVtcHR5LXBhdHRlcm4iOjIsIm5vLWV4LWFzc2lnbiI6Miwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjoyLCJuby1leHRyYS1zZW1pIjoyLCJuby1mYWxsdGhyb3VnaCI6Miwibm8tZnVuYy1hc3NpZ24iOjIsIm5vLWdsb2JhbC1hc3NpZ24iOjIsIm5vLWltcG9ydC1hc3NpZ24iOjIsIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6Miwibm8taW52YWxpZC1yZWdleHAiOjIsIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjoyLCJuby1sb3NzLW9mLXByZWNpc2lvbiI6Miwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6Miwibm8tbmV3LXN5bWJvbCI6Miwibm8tbm9ub2N0YWwtZGVjaW1hbC1lc2NhcGUiOjIsIm5vLW9iai1jYWxscyI6Miwibm8tb2N0YWwiOjIsIm5vLXByb3RvdHlwZS1idWlsdGlucyI6Miwibm8tcmVkZWNsYXJlIjoyLCJuby1yZWdleC1zcGFjZXMiOjIsIm5vLXNlbGYtYXNzaWduIjoyLCJuby1zZXR0ZXItcmV0dXJuIjoyLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6Miwibm8tc3BhcnNlLWFycmF5cyI6Miwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOjIsIm5vLXVuZGVmIjoyLCJuby11bmV4cGVjdGVkLW11bHRpbGluZSI6Miwibm8tdW5yZWFjaGFibGUiOjIsIm5vLXVuc2FmZS1maW5hbGx5IjoyLCJuby11bnNhZmUtbmVnYXRpb24iOjIsIm5vLXVuc2FmZS1vcHRpb25hbC1jaGFpbmluZyI6Miwibm8tdW51c2VkLWxhYmVscyI6Miwibm8tdW51c2VkLXZhcnMiOjIsIm5vLXVzZWxlc3MtYmFja3JlZmVyZW5jZSI6Miwibm8tdXNlbGVzcy1jYXRjaCI6Miwibm8tdXNlbGVzcy1lc2NhcGUiOjIsIm5vLXdpdGgiOjIsInJlcXVpcmUteWllbGQiOjIsInVzZS1pc25hbiI6MiwidmFsaWQtdHlwZW9mIjoyfSwiZW52Ijp7fX19) that shows that the error _should_ be happening (confirming our understanding of the rule itself)
```
```javascript
// both of these functions error in the playground
function bar(baz) {
}

function bar2(baz) {
  baz = baz || {};
}
var foo = 5;
bar();
bar2(foo);
```

I commented out our configuration for `'@typescript-eslint/no-unused-vars'` _just_ in case, and that didn't yield any issues.  So my best guess is either
1. We need to tweak our `'@typescript-eslint/no-unused-vars'` configuration in some way
    1. The rule does suggest turning off the ESLint version, which we don't do today. I added it back locally and couldn't get it to fire on a new (unused) parameter for the function
3. There's an issue (or maybe by design) that no-unused-vars in TS handles this case differently

Tl;dr :shrug:

Comment (alicewriteswrongs) [reply src/compiler/transformers/test/transpile.ts:13]: hmmm well sometimes things are just mysteries 🤷‍♀️
